src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/Commands/MainListPage.cs

```diff
@@ -131,4 +131,39 @@ private void StartAppLoad()
             IsLoading = ActuallyLoading();
         });
     }
+
+    // Almost verbatim ListHelpers.ScoreListItem, but also accounting for the
+    // fact that we want fallback handlers down-weighted, so that they don't
+    // _always_ show up first.
+    private static int ScoreTopLevelItem(string query, IListItem topLevelOrAppItem)
+    {
+        if (string.IsNullOrEmpty(query))
```

Comment:
Just thinking through this -- for fallbacks that don't want to show up unless some certain conditions are met... are those items just not even returned from the CommandsProvider before this step, so the ranking doesn't really matter here of 1? 

At first I was thinking "oh well if this check is first, does that mean all fallbacks show up on empty string?" and it feels like most (all?) fallbacks should not show up on the empty string search. But now, I'm thinking that the fallback commands that want specific matches for strings won't even get to this stage?